storage/imx6ull-flash/imx6ull-flash-test.c

```diff
@@ -1006,6 +1006,73 @@ void test_single_block_raw(flashdrv_dma_t *dma, uint32_t blockno, uint8_t *data,
 	TIMEPROF_END_WARN(100, "erase(3)");
 }
 
+/* write, read and check */
+void _write_and_check(flashdrv_dma_t *dma, uint32_t blockno, uint8_t *data, uint8_t *meta, uint8_t byte)
+{
+	uint32_t addr = blockno * 64;
+	unsigned int i;
+	int err;
+	flashdrv_meta_t *aux = (flashdrv_meta_t *)meta;
+
+	memset(data, byte, pagemapsz);
+	for (i = 0; i < sizeof(aux->metadata); ++i) {
+		aux->metadata[i] = i;
+	}
+
+	if ((err = flashdrv_write(dma, addr, data, (char *)meta)) < 0)
+		printf("[%4u] write(0x%02x) failed: %d\n", blockno, byte, err);
+
+	memset(data, 0, pagemapsz);
+	if ((err = flashdrv_read(dma, addr, data, aux)) < 0)
+		printf("[%4u] read(0x%02x) failed: %d\n", blockno, byte, err);
+
+	for (i = 0; i < sizeof(aux->metadata); i++) {
+		if (aux->metadata[i] != i)
+			printf("[%4u] meta[%2u] invalid data 0x%02x\n", blockno, i, aux->metadata[i]);
+	}
+
+	for (i = 0; i < sizeof(aux->errors); i++) {
+		switch (aux->errors[i]) {
+			case flash_no_errors:
+			case flash_erased:
+				break;
+
+			case flash_uncorrectable:
+				printf("[%4u] uncorrectable data in chunk %u\n", blockno, i);
+				break;
+
+			default:
+				printf("[%4u] %u corrections in chunk %u\n", blockno, aux->errors[i], i);
+		}
+	}
+
+	for (i = 0; i < flashinfo.writesz; ++i) {
+		if (data[i] != byte)
+			printf("[%4u] write/read(0x%02x)[%u] invalid data: 0x%02x\n", blockno, byte, i, data[i]);
+	}
+}
+
+void test_single_block(flashdrv_dma_t *dma, uint32_t blockno, uint8_t *data, uint8_t *meta)
```

Comment:
I know this is aligned with how tests are written currently but I would have preferred for test to return if it passed or failed.

To be considered in the future.

Reply:
ack